app/controllers/notices_controller.rb

```diff
@@ -67,8 +67,15 @@ def show
         end
 
         format.json do
-          render json: @notice, serializer: NoticeSerializerProxy,
-          root: json_root_for(@notice.class)
+          if request.headers['HTTP_AUTHENTICATION_TOKEN'] && User.find_by_authentication_token(request.headers['HTTP_AUTHENTICATION_TOKEN']).has_role?(Role.researcher)
+            render json: @notice, serializer: NoticeSerializerProxy,
+            root: json_root_for(@notice.class)
+            Rails.logger.info "It worked."
+          else
+            render json: @notice, serializer: LimitedNoticeSerializerProxy,
+            root: json_root_for(@notice.class)
+            Rails.logger.info "Go fish!"
+          end
         end
       end
     end
```

Comment:
How does this strike you:
```ruby
      serializer = if researcher? ? NoticeSerializerProxy : LimitedNoticeSerializerProxy
      render json: @notice, serializer: serializer, root: json_root_for(@notice.class)
    .
    .
    .

    private

    def researcher?
      return false unless request.headers['HTTP_AUTHENTICATION_TOKEN']
      User.find_by_authentication_token(
        request.headers['HTTP_AUTHENTICATION_TOKEN']
      ).has_role?(Role.researcher)
    end
```
Also, quick question, is current_user not available here? If it is, would something like this work?
```ruby
    def researcher?
      current_user && current_user.has_role?(Role.researcher)
    end
```